homeassistant/components/shell_command/__init__.py

```diff
@@ -54,7 +60,7 @@ async def async_service_handler(service: ServiceCall) -> None:
                 )
             except TemplateError as ex:
                 _LOGGER.exception("Error rendering command template: %s", ex)
-                return
+                return None
```

Comment:
Can we add a test covering this?

Reply:
While working on this I found that the following bit in the core `async_call` prevents `ServiceResponse` from being returned as `None`:
https://github.com/home-assistant/core/blob/dc8267b05a5c81c484b10d0ab11daf88cbdfdfe4/homeassistant/core.py#L1978-L1982

That seems counter intuitive to me.

@allenporter, Seeing as you've implemented this mechanic. Could you elaborate on the implementation pointed to?
IMO if something goes wrong during a service call, returning None should be possible and not be captured again in the core's `async_call`.

Reply:
If something goes wrong an exception should be raised.

Reply:
See https://developers.home-assistant.io/docs/dev_101_services?_highlight=service&_highlight=respo#response-data for our design standards agreed upon in the architecture discussion.

Reply:
Thanks for the info.
The exeptions in this case are handled and don't propagate. To be true to the design standards I believe these exceptions should then be re-raised. Though to remain consistent with the existing implementation of `shell_command` we should stick to the logging produced and return gracefully.

I lean towards the latter, but Ok with either implementation.

Reply:
I don't think the existing implementation is right either. This should be raising instead of swallowing even if not considering response values. The response values guidance is reiterating the current desired best practice.